src/components/Table/TableSaveViewModal/TableSaveViewModal.test.jsx

```diff
@@ -0,0 +1,260 @@
+
+    expect(actions.onSave).toHaveBeenCalledWith({
+      isDefault: false,
+      isPublic: false,
+      title: 'testval1',
+    });
+
+    fireEvent.change(getByTestId('my-modal-form-title-input'), { target: { value: 'testval2' } });
+    fireEvent.click(getByLabelText(i18n.defaultCheckboxLabelText));
+    fireEvent.click(getByLabelText(i18n.publicCheckboxLabelText));
+    fireEvent.click(getByText(i18n.saveButtonLabelText));
+
+    expect(actions.onSave).toHaveBeenCalledWith({
+      isDefault: true,
+      isPublic: true,
+      title: 'testval2',
+    });
+  });
+
+  it('calls onChange when form elements value changes', () => {
+    const { getByTestId, getByLabelText } = render(
+      <TableSaveViewModal actions={actions} open testID="my-modal" />
+    );
+    fireEvent.change(getByTestId('my-modal-form-title-input'), { target: { value: 'testval1' } });
+    fireEvent.click(getByLabelText(i18n.defaultCheckboxLabelText));
+    fireEvent.click(getByLabelText(i18n.publicCheckboxLabelText));
+
+    expect(actions.onChange).toHaveBeenCalledWith({
+      title: 'testval1',
+    });
+    expect(actions.onChange).toHaveBeenCalledWith({
+      isDefault: true,
+    });
+    expect(actions.onChange).toHaveBeenCalledWith({
+      isPublic: true,
+    });
+  });
+
+  it('calls onClose on cancel and close-icon click', () => {
+    const { getByRole, getByText } = render(
+      <TableSaveViewModal actions={actions} open testID="my-modal" />
+    );
+    fireEvent.click(getByText(i18n.cancelButtonLabelText));
+    fireEvent.click(getByRole('button', { name: i18n.closeIconDescription }));
+
+    expect(actions.onClose).toHaveBeenCalledTimes(2);
+  });
+
+  it('handles errors correctly', () => {
+    const { getByRole, queryByText } = render(
+      <TableSaveViewModal actions={actions} open testID="my-modal" error="my test error" />
+    );
+    expect(queryByText('my test error')).toBeInTheDocument();
+    expect(actions.onClearError).not.toHaveBeenCalled();
+
+    fireEvent.click(getByRole('button', { name: 'closes notification' }));
+    expect(queryByText('my test error')).not.toBeInTheDocument();
+    expect(actions.onClearError).toHaveBeenCalled();
+  });
+
+  it('replaces the strings with i18n strings', () => {
+    const i18nTest = {
+      modalTitle: 'modal-title-test',
+      modalBodyText: 'modal-body-test',
+      titleInputLabelText: 'title-input-label-test',
+      defaultCheckboxLabelText: 'default-checkbox-label-test',
+      publicCheckboxLabelText: 'public-checkbox-label-test',
+      closeIconDescription: 'close-icon-description-test',
+      saveButtonLabelText: 'save-button-label-test',
+      cancelButtonLabelText: 'cancel-button-label-test',
+    };
+
+    const {
+      getByText,
+      getByLabelText,
+      queryByText,
+      queryByLabelText,
+      getByTitle,
+      queryByTitle,
+    } = render(<TableSaveViewModal actions={actions} open testID="my-modal" i18n={i18nTest} />);
```

Comment:
you can avoid this block by using `screen` instead. It's a best practice to aim for moving forward.

import `screen` from `@testing-library/react` above and then:

```suggestion
render(<TableSaveViewModal actions={actions} open testID="my-modal" i18n={i18nTest} />);
```

and then in your queries it's just `expect(screen.getByWhatever(thing)).toBeWhatever();`

Reply:
Awesome, thanks! I will change it.